### PR TITLE
Remove "let f => x" support.

### PR DESCRIPTION
**Pull request by SanderSpies**
#1051

### CODE REVIEW

**Comment by kennetpostigo** [comment]
@SanderSpies I've used this everywhere in my guide. What should I use instead?

**Comment by jordwalke** [comment]
@kennetpostigo Are you sure you used this exact form? Notice that there is no argument (and I'm just now finding out that it is parsed as a single `()` argument). I had no idea this existed!

**Comment by kennetpostigo** [comment]
Sorry, I misinterpreted the PR. This has to do with `unit` being returned implicitly right?

**Comment by yawaramin** [comment]
@kennetpostigo it does have to do with function expression syntax--not `unit` being returned implicitly. Specifically,

```ocaml
let f => x;
```

is being interpreted as

```ocaml
let f () => x;
```

This is tripping people up.

**Comment by kennetpostigo** [comment]
@yawaramin Interesting, thank you for explaining!

File: src/reason_parser.messages [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# ------------------------------------------------------------------------------
+©# ------------------------------------------------------------------------------
```
Comment (chenglou) [inline src/reason_parser.messages]: @SanderSpies lol?

**Comment by chenglou** [comment]
Now passes. Merging! Thanks.

File: src/reason_parser.messages (hunk 1 above)
Comment (SanderSpies) [inline src/reason_parser.messages]: Yikes.

**Pull request merged**